lib/src/webcrypto/webcrypto.ecdh.dart

```diff
@@ -14,43 +14,250 @@
+  ///   final exportedJsonWebKey = await jsonWebKey.exportJsonWebKey();
+  ///   print(exportedJsonWebKey);
+  /// }
+  /// ```
+  /// 
+  /// [1]: https://www.rfc-editor.org/rfc/rfc7518.html#section-6.2
   static Future<EcdhPrivateKey> importJsonWebKey(
     Map<String, dynamic> jwk,
     EllipticCurve curve,
   ) {
     return impl.ecdhPrivateKey_importJsonWebKey(jwk, curve);
   }
 
+  /// Generate a new [EcdhPrivateKey] and [EcdhPublicKey] pair.
+  ///
+  /// The [curve] parameter specifies the curve to use for the key pair.
+  /// 
+  /// Use [EllipticCurve.p256] for the P-256 curve.
+  /// Use [EllipticCurve.p384] for the P-384 curve.
+  /// Use [EllipticCurve.p521] for the P-521 curve.
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Export the private key.
+  ///   final exportedPrivateKey = await keyPair.privateKey.exportJsonWebKey();
+  ///   print(exportedPrivateKey);
+  /// 
+  ///   // Export the public key.
+  ///   final exportedPublicKey = await keyPair.publicKey.exportJsonWebKey();
+  ///   print(exportedPublicKey);
+  /// }
+  /// ```
+  /// 
   static Future<KeyPair<EcdhPrivateKey, EcdhPublicKey>> generateKey(
     EllipticCurve curve,
   ) {
     return impl.ecdhPrivateKey_generateKey(curve);
   }
 
+  /// Derive an array of bits from the [EcdhPrivateKey].
+  ///
+  /// The [length] parameter specifies the number of bits to derive and 
+  /// should be multiples of 8.
+  /// The [publicKey] parameter is the [EcdhPublicKey].
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Derive 256 bits from the private key using the public key.
+  ///   final derivedBits = await keyPair.privateKey.deriveBits(256, keyPair.publicKey);
+  ///   print(derivedBits);
+  /// }
+  /// ```
+  Future<Uint8List> deriveBits(int length, EcdhPublicKey publicKey);
   // Note some webcrypto implementations (chrome, not firefox) supports passing
   // null for length (in this primitive). However, you can always know the right
   // length from the curve. Note p512 can provide up to: 528 bits!!!
   //
   // See: https://tools.ietf.org/html/rfc6090#section-4
   // Notice that this is not uniformly distributed, see also:
   // https://tools.ietf.org/html/rfc6090#appendix-B
-  Future<Uint8List> deriveBits(int length, EcdhPublicKey publicKey);
 
-  // Note. unsupported on Firefox, see EcdsaPrivateKey.importPkcs8Key
+  /// Export the [EcdhPrivateKey] as a PKCS8 key.
+  ///
+  /// This returns the private key as an octet string.
```

Comment:
I would just say: returns private key as a list of bytes.

Unless, we're really fancy. Look at `RsassaPkcs1V15PrivateKey.exportPkcs8Key` it says:
> Returns the DER encoding of the PrivateKeyInfo structure specified in [RFC 5208](https://tools.ietf.org/html/rfc5208) as a list of bytes.

I assume I really did look up the specification to find out exactly what is returned and in what format.

I'm guessing it might be a different thing for EC keys.

--------------

I actually think I did this, because normally when someone says "PKCS #8" format, you're thinking about a base64 encoded string inside the PEM `---- PRIVATE KEY---` ASCII armor thing..

Which is not what this returns.

Look at: https://www.w3.org/TR/WebCryptoAPI/#ecdh-operations

So it might be worth saying what structure it returns in what encoding as bytes.

I suspect this is also where we have to at-least for our own sanity read about BER vs DER (I forgot all about this).
See: https://en.wikipedia.org/wiki/X.690

But I think webcrypto might only support DER encoding (this might be something we should test across platforms, probably something we should file an issue for testing :rofl: )

--------------------

So, yeah, I'm not exactly sure what this should say. We should probably read a bit more in the RFCs and webcrypto spec to make sure what references we want to give.